tests/test_tree_kem_kat.rs

```diff
@@ -0,0 +1,278 @@
+//! TreeKEM test vectors
+//!
+//! See https://github.com/mlswg/mls-implementations/blob/master/test-vectors.md
+//! for more description on the test vectors.
+//!
+//! The test vector describes a tree of `n` leaves adds a new leaf with
+//! `my_key_package` and `my_path_secret` (common ancestor of `add_sender` and
+//! `my_key_package`).
+//! Then an update, sent by `update_sender` with `update_path` is processed, which
+//! is processed by the newly added leaf as well.
+//!
+//! Some more points
+//! * An empty group context is used.
+//! * update path with empty exclusion list.
+
+use openmls::{prelude::*, test_util::*};
+
+mod utils;
+use utils::managed_utils::*;
+
+use serde::{self, Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct TreeKemTestVector {
```

Comment:
This file contains a lot of duplicated code. We should find a way to share code that's used in both, generation and test.

Reply:
Good point. I've made the values in the "internal" test vector definition public such that we can use it here.